src/main/java/seedu/address/ui/AddWindow.java

```diff
@@ -165,19 +188,60 @@ private void handleAdd() {
 
         try {
             executeCommand(userInput.toString());
-            // reset all fields and then hide the panel
-            this.resetFields();
-            this.hide();
         } catch (CommandException | ParseException e) {
             errorLabel.setText("Error encountered");
+            return;
+        }
+
+        // Since user command execution is successful, then we do the other stuff next.
+        // Notice that whenever a new Person is added into AddressBook, it'll list out all Persons.
+        // So we simply need to retrieve the last Person added...
+        ReadOnlyAddressBook ab = logic.getAddressBook();
+        ObservableList<Person> personList = ab.getPersonList();
+        int lastIndex = personList.size();
+
+        // Do not need to handle the fact that the given status might not be valid.
+        if (!status.equals("")) {
+            String setStatus = "status " + lastIndex + " " + status;
+            try {
+                executeCommand(setStatus);
+            } catch (CommandException | ParseException e) {
+                errorLabel.setText("Error encountered\nEnsure status is only blacklist, whitelist, or empty.");
+                return;
+            }
+        }
```

Comment:
Could add SLAP for  handling status and module? Probably unnecessary.

Reply:
Fixed in latest commit, thanks!